src/librbd/internal.cc

```diff
@@ -1570,9 +1570,16 @@ void filter_out_mirror_watchers(ImageCtx *ictx,
 
     managed_lock::Locker locker;
     C_SaferCond get_owner_ctx;
-    auto get_owner_req = managed_lock::GetLockerRequest<>::create(
-      ictx->md_ctx, ictx->header_oid, &locker, &get_owner_ctx);
-    get_owner_req->send();
+    {
+      RWLock::RLocker l(ictx->owner_lock);
+
+      if (ictx->exclusive_lock == nullptr) {
```

Comment:
Then I suppose I can use `ExclusiveLock(*ictx).get_locker(&locker, &get_owner_ctx)` for this case, still looks a little better.